### PR TITLE
Add TimeRange.contains(t) method

### LINKED ISSUE #3789: Document t in TimeRange
`TimeRange` had a `.__contains__` method, but this isn't well documented. We should add some more obvious documentation that one can do `t in TimeRange`.

### PR DESCRIPTION
**Pull request by dstansby**
Potentially fixes #3789

### CODE REVIEW

**Comment by nabobalis** [comment]
I wonder if we need to import `autodoc_default_options` first.

**Comment by nabobalis** [comment]
To see it working nicely: https://39302-2165383-gh.circle-artifacts.com/0/docs/_build/html/api/sunpy.time.TimeRange.html?highlight=timerange

**Comment by wtbarnes** [comment]
Does this need to be documented as an explicit method on `TimeSeries`? It seems like this would encourage people to use the `contains`  method directly. There's not anything wrong with that *per se*, but it seems to defeat the purpose of the `__contains__`...

Instead of adding this to the public API for the sake of documentation, could we not just add an example of how the `__contains__` method is used (i.e. the example with `this_time in that_time`) to the main `TimeSeries` docstring?

**Comment by Cadair** [comment]
I am also kinda +0 on adding a new method which does the same thing. The big advantage of doing this to me is tab complete, but at the same time it feels redundant. I agree with @wtbarnes that isn't there a better way we can document this?

**Comment by nabobalis** [comment]
> I am also kinda +0 on adding a new method which does the same thing. The big advantage of doing this to me is tab complete, but at the same time it feels redundant. I agree with @wtbarnes that isn't there a better way we can document this?

If you have one?

File: sunpy/time/timerange.py [hunk 1]
```diff
@@ -405,7 +405,8 @@ def __contains__(self, time):
         """
         Checks whether the given time lies within this range. Both limits are
```
Comment (Cadair) [inline sunpy/time/timerange.py]: We should modify this docstring to be more logical for `.contains()` and also to demonstrate the `t in tr` syntax as well.

**Comment by dstansby** [comment]
For the documentation was the initial motivation, but I also think doing `TimeRange.contains(t)` is slghtly more obvious syntax than `t in TimeRange`.

Consensus seems to be to add documentation to the docstring of `TimeRange` though instead of adding `.contains()`, so I will do that and open an alternative PR.

**Pull request closed without merging**